File: 41-first-missing-positive/first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int n = nums.length;
        for(int i=0;i<n;i++){
            while( nums[i]>0 &&
                nums[i]<=n &&
                nums[i] != nums[nums[i]-1]
            ){
            int cind = nums[i] -1;
            //swap
            int temp = nums[i];
            nums[i] = nums[cind];
            nums[cind] = temp;
            }
            
        }
        for(int i=0;i<n;i++){
            if(i+1 !=nums[i]) return i+1;
        }  
        return n +1;      
    }
}

File: 41-first-missing-positive/first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n = nums.size();
        int idx = 0;
        for(int i =0;i<n;i++){
            if(nums[i]> 0){
                idx = i;
                break;
            }
        }
        int sum  = 0;
        int count = 1;
        
        for (int i = idx ;i<n;i++){
            if (i > idx && nums[i] == nums[i - 1]) {
                continue;
            }
            if(nums[i] == count)
                count++;
            else if(nums[i]> count) {
                return count;
            }
        }
        return count;
    }
};